File: android/src/main/kotlin/com/example/design_engine/layer2_core/light/EOnPrimaryFixedLight.kt
package com.example.design_engine.layer2_core.light
import com.example.design_engine.R
import com.example.design_engine.layer3_logic.AppTheme
object EOnPrimaryFixedLight {
    fun getResId(activeTheme: AppTheme): Int {
        return when (activeTheme) {
            AppTheme.SYSTEM -> R.string.csyl22
            AppTheme.STANDARD -> R.color.cstl22
            AppTheme.BLUE -> R.color.cbl22
            AppTheme.GREEN -> R.color.cgl22
            AppTheme.RED -> R.color.crl22
            AppTheme.NOTHING -> R.color.cnol22
        }
    }
}

File: lib/layer2_core/light/e_on_primary_fixed_light.dart
import '../../layer1_raw/design_token.dart';
import '../../layer1_raw/light/system.dart';
import '../../layer1_raw/light/standard.dart';
import '../../layer1_raw/light/blue.dart';
import '../../layer1_raw/light/green.dart';
import '../../layer1_raw/light/red.dart';
import '../../layer1_raw/light/nothing.dart';

class EOnPrimaryFixedLight {
  static DesignToken getToken(dynamic activeTheme) {
    switch (activeTheme.toString().split('.').last) {
      case 'system': return csyl22;
      case 'standard': return cstl22;
      case 'blue': return cbl22;
      case 'green': return cgl22;
      case 'red': return crl22;
      case 'nothing': return cnol22;
      default: return cstl22;
    }
  }
}
